server/auth.js

```diff
@@ -33,6 +35,32 @@ exports.login = async function (username, password) {
     return null;
 };
 
+/**
+ * Validate a provided API key
+ * @param {string} key API Key passed by client
+ * @returns {Promise<bool>}
+ */
+async function validateAPIKey(key) {
+    if (typeof key !== "string") {
+        return false;
+    }
+
+    let index = key.substring(0, key.indexOf("-"));
+    let clear = key.substring(key.indexOf("-") + 1, key.length);
+    console.log(index);
+    console.log(clear);
+
+    let hash = await R.findOne("api_key", " id=? ", [ index ]);
+
+    let current = dayjs();
+    let expiry = dayjs(hash.expires);
+    if (expiry.diff(current) < 0, !hash.active) {
```

Comment:
It should be or (`||`) here?

Reply:
Yes. Looking into why the comma didn't throw a syntax error, apparently it is valid and just causes only the last expression to be used for the condition, which explains why it still works